jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,843 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize Science Data with Cubeviz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Cubeviz incompatible with jupyter_client 6.1.6.  Use jupyter_client 5.3.5 instead."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "cubeviz = CubeViz()\n",
+    "cubeviz.app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### UI Instructions:\n",
+    "* Load science datacube into Cubeviz using the cell below\n",
+    "* Go to the Hammer-and-Screwdriver icon: Gear icon: Layer in the leftmost image viewer \n",
+    "* In that tab, change the Linear stretch to 90 percentile to see the faint QSO target at (x,y) ~ (17, 21)\n",
+    "* Scrubbing through the cube also helps to locate the source\n",
+    "* Select a circular subset region centered on the source. \n",
+    "* Note that the region is pixelated and doesn't include fractional pixels\n",
+    "* Change the collapse method to \"Sum\" in spectrum viewer: Gear icon : Viewer \n",
+    "* --This \"Sum\" method yields our subset extraction\n",
+    "* Change the vertical zoom to see the spectral features in the Subset spectrum"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Notes*: \n",
+    "\n",
+    "(1) Image viewer contrast settings change when you click on the side bar to expand/contract jupyter scroll window\n",
+    "\n",
+    "(2) Spectrum viewer: Viewer cube collapse method should default to Sum (not Maximum)\n",
+    "\n",
+    "(3) Spectrum viewer y scale returns to autoscale when the region is moved, and y-zoom has to be adjusted again\n",
+    "\n",
+    "(4) Region selection appears away from cursor after opening hammer-and-screwdriver to change cube viewer contrast"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load Cube into Cubeviz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Data from local directory\n",
+    "#cubeviz.app.load_data(filename)\n",
+    "\n",
+    "#Data from url:\n",
+    "url = filename\n",
+    "df = download_file(url)\n",
+    "time.sleep(2)  # Sleep to avoid glue-jupyter timing issue\n",
+    "cubeviz.app.load_data(df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Developer Note:** Spectral cube does not yet recognize JWST NIRSpec IFU datacubes, giving the above warning\n",
+    "for each FITS extension."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Export Region from Cubeviz\n",
+    "Export the region defined by the user in Cubeviz as an astropy CirclePixel Region, which has units of pixels."
```

Comment:
Here I would add. "See the instructions under the Cubeviz viewer on how to select the regions of interest," as it stands, it may be confusing to the user.

Reply:
Maybe the loading cell should come before these instructions? We could add, "now that the data has been loaded, scroll up to apply the following: ...<instructions>"

Reply:
Disagree.